File: goodPair.go
package main

import "fmt"

func main() {
	inputArr := []int{1, 2, 3, 4}
	fmt.Println(solve(inputArr, 66))
}

func solve(A []int, B int) int {
	goodPairCount := 0
	for i := 0; i < len(A); i++ {
		for j := i + 1; j < len(A); j++ {
			if i != j && (A[i]+A[j] == B) {
				goodPairCount += 1
			}
		}
	}
	return goodPairCount
}

File: goodpair.py
def solve(A, B):
    hashmap = {}

    for i in A:
        if i in hashmap:
            hashmap[i] += 1
        else:
            hashmap[i] = 1

    for i in hashmap:
        if B-i in hashmap:
            if i*2 != B:
                return 1
            if i*2 == B and hashmap[i] > 1:
                return 1
    return 0


A = [1, 2, 2]
C = len(A)
B = 4
print(solve(A, B))
